lib/msal-browser/scripts/README.md

```diff
@@ -0,0 +1,97 @@
+# MSAL Browser Log Decoder
+
+This script decodes hashed logging strings from browser console logs back to their original messages.
+
+## Overview
+
+The MSAL Browser library hashes logging strings to reduce bundle size. When running in a browser, console logs display these hashes instead of full messages. This decoder script transforms those hashed logs back into readable messages and saves them to a new file.
+
+## Log Format
+
+The script expects log files where each MSAL log entry follows this format:
+```
+[timestamp] : [correlation-id] : @azure/[module]@[version] : [LogLevel] - [hash]
+```
+
+Example:
+```
+[Tue, 07 Oct 2025 16:50:29 GMT] : [] : @azure/msal-browser@4.13.1 : Verbose - 0hoqeo
+[Tue, 07 Oct 2025 16:50:30 GMT] : [abc-123] : @azure/msal-common@15.7.0 : Info - 1atvtd
+```
+
+Non-MSAL log lines are preserved unchanged in the output.
+
+## Usage
+
+### Basic Usage
+```bash
+node scripts/decode-logs.cjs <path-to-log-file>
+```
+
+### With NPM Script
+```bash
+npm run decode-logs -- <path-to-log-file>
+```
+
+### Verbose Mode
+Add `--verbose` or `-V` flag for detailed debugging information:
+```bash
+node scripts/decode-logs.cjs --verbose <path-to-log-file>
+npm run decode-logs -- --verbose /path/to/console.log
+```
+
+### Examples
+```bash
+# Decode logs from a file
+node scripts/decode-logs.cjs ./console-logs.txt
+# Output: ./console-logs-decoded.txt
+
+# Decode with verbose output
+node scripts/decode-logs.cjs --verbose /Users/user/Downloads/browser-console.log
+# Output: /Users/user/Downloads/browser-console-decoded.log
+
+# Using npm script
+npm run decode-logs -- ./logs/debug.log
+# Output: ./logs/debug-decoded.log
+```
+
+## Output
+
+The script creates a new file with the decoded logs in the same directory as the input file, adding a `-decoded` suffix before the file extension:
+
+- Input: `/path/to/debug.log` → Output: `/path/to/debug-decoded.log`
+
+## Features
+
+- **Automatic Version Detection**: Extracts module@version from each log entry and fetches appropriate mappings
+- **Multi-Module Support**: Handles logs from multiple MSAL packages (msal-browser, msal-common, etc.)
+- **Multiple Versions Support**: Can handle multiple versions of the same module in a single file
```

Comment:
How will this look like? What will be the use case for multiple versions, as in, is the assumption that a user combines the logs from different msal js sessions?